server/app.js

```diff
@@ -18,10 +21,37 @@ app.use(function(req, res, next) {
 });
 
 app.get('/books/:id', (req, res) => {
+  console.log("Got request for id:", req.params.id);
   books.retrieve(req.params.id, (err, doc) => {
     // add err handling
+    console.log("sending response :", doc, err);
     res.send(doc);
   });
 });
 
+app.post('/books', (req, res) => {
+  console.log("Post request for:");
+  res.send("response for Post");
+});
+
+app.delete('/books/:id', (req, res) => {
+  console.log("Delete request for id:", req.params.id);
+  res.send("response for Delete");
+});
+
+app.put('/books', (req, res) => {
+  console.log("Update request for:", req.body);
+  res.send("response for Update");
+});
+
+
+app.all('/books',
+  errorHandler.httpError(405));
+
+
+// Deliver 404 errors for any unhandled routes.
+// Express has a 404 handler built-in, but it
+// won't deliver errors consistent with your API.
```

Comment:
What do you mean by "consistent with our API"?, I would elaborate more in the comments